### ISSUE TITLE
Old IE doesn't pass referer when using window.location

### ISSUE THREAD

**Issue by stiefdirckx** [JimBobSquarePants/jQuery-Google-Analytics#2]
You open the click events delayed to give GA time to push the event. But when using window.location, IE 8 and below don't pass the referer to the next page. This can cause problems with analytics on external sites. I changed your code to use a small trick:
http://stackoverflow.com/questions/4762254/javascript-window-location-does-not-set-referer-in-the-request-header

I also included a small feature to open links in a new window when the link has a target with value '_blank'.

setTimeout(function () {
    var href = self.$element.attr("href");
    var target = self.$element.attr("target");

```
if (href && href.indexOf("#") !== 0) {
    if (target == "_blank") {
        window.open(href, target);
    }
    else {
        var a = document.createElement("a");

        if (!a.click) {
            window.location = href;
            return;
        }

        a.setAttribute("href", url);
        a.style.display = "none";
        document.body.appendChild(a);
        a.click();
    }
}
```

}, 100);

Maybe you would like to add it to your solution also?

**Comment by JimBobSquarePants** [comment]
Thanks for that, I wasn't aware of the issue in IE8. Fixed now.

**Comment by stiefdirckx** [comment]
I did some further testing with passing the referrer to external sites and there still where some situations where IE didn't pass it when the link was opening in an external window. I rearranged some code and now it works in Chrome, IE and Firefox I think.

'if (!a.click)' is only used now for (very) old Firefox versions. For simplicity you could remove the code for opening in external window for this rare situation. All recent browsers use the pseudo created A-tag so I've moved the target check there.

``` javascript
if (href && href.indexOf("#") !== 0) {

    // IE8 doesn't pass the referrer to the next page
    // using window.location. We create a hidden link to 
    // counter that. Fixes issue #2
    // http://stackoverflow.com/a/7917528/427899
    var a = document.createElement("a");

    if (!a.click) {
        // Handle external window requests.
        if (target && target === "_blank") {
            window.open(href, target);
            return;
        }
        else {
            // Old FF and Secure links.
            window.location = href;
            return;
        }
    }

    a.setAttribute("href", href);
    if (target && target === "_blank") {
        a.setAttribute("target", target);
    }
    a.style.display = "none";
    document.body.appendChild(a);
    a.click();
}
```

**Comment by JimBobSquarePants** [comment]
Interesting. I'll get it updated asap. Thanks for all this testing btw. I've not had the opportunity other than last night to have a look at it in a long while
